Feature/splash&onboarding

ASHmd97 opened the pull request.

ASHmd97 committed:
Add onboarding feature with new assets and localization support

ASHmd97 committed:
Add splash screen assets and update launch themes for Android and iOS

ASHmd97 committed:
Add login page and update routing for onboarding flow

ASHmd97 committed:
Add onboarding page tests with navigation and button interactions

john-safwat commented on flutter_native_splash.yaml:
this code handle only the Android 12+ and building splash screen as image isn't the correct way

john-safwat commented on lib/features/onboarding/presentation/pages/on_boarding_page.dart:
```diff
@@ -0,0 +1,87 @@
+
+  @override
+  State<OnBoardingPage> createState() => _OnBoardingPageState();
+  final AppSecureStorage localStorageClient = getIt<AppSecureStorage>();
```
local storage is a local data source accessing here is a violation of the clean arch

john-safwat commented on lib/features/onboarding/presentation/pages/on_boarding_page.dart:
```diff
@@ -0,0 +1,87 @@
+    super.initState();
+    // Check if the user has already seen the onboarding
+    // widget.localStorageClient.saveBool(LocalConstants.onBoardingKey, true);
+  }
```
since this code is commented you should remove it before uploading it

john-safwat commented on lib/features/onboarding/presentation/pages/on_boarding_page.dart:
```diff
@@ -0,0 +1,87 @@
+              children: [
+                Expanded(
+                  child: PageView(
+                    physics: const NeverScrollableScrollPhysics(),
```
user can scroll no need for this physics

john-safwat commented on lib/features/onboarding/presentation/pages/on_boarding_page.dart:
```diff
@@ -0,0 +1,87 @@
+                        imagePath: Assets.imagesO153,
+                        title: context.l10n.onBoardingTitle3,
+                        description: context.l10n.onBoardingDesc3,
+                      ),
```
miss leading images name

john-safwat commented on lib/features/onboarding/presentation/widgets/custom_elevated_btn.dart:
```diff
@@ -0,0 +1,36 @@
+      style: ElevatedButton.styleFrom(
+        padding: EdgeInsets.symmetric(horizontal: 16, vertical: 8),
+
+        backgroundColor: Color(0xffFF4100),
```
don't use hard coded colors in your code

john-safwat commented on lib/features/onboarding/presentation/widgets/custom_elevated_btn.dart:
```diff
@@ -0,0 +1,36 @@
+        shape: RoundedRectangleBorder(
+          borderRadius: BorderRadius.circular(20.sp),
+        ),
+      ),
```
elevated button style should be in the app theme

john-safwat commented on lib/features/onboarding/presentation/widgets/custom_elevated_btn.dart:
```diff
@@ -0,0 +1,36 @@
+          color: Colors.white,
+          fontSize: 14,
+          fontWeight: FontWeight.w800,
+        ),
```
use text style from app theme don't use hard coded styles

john-safwat commented on lib/features/onboarding/presentation/widgets/custom_elevated_btn.dart:
creating custom elevated button for changing the style is not acceptable you should add this style in app theme

john-safwat commented on lib/features/onboarding/presentation/widgets/custom_outlined_btn.dart:
creating custom Outlined button for changing the style is not acceptable you should add this style in app theme

john-safwat commented on lib/features/onboarding/presentation/widgets/page_view_item.dart:
this is a complex code and not accurate for all screens refactor this code using expanded

john-safwat commented on lib/features/onboarding/presentation/widgets/page_view_item.dart:
don't give buttons static hight using sized box

john-safwat reviewed (changes requested):
this code need many changes and more test coverage

ASHmd97 committed:
fix(onboarding): apply all code review

Refactored UI code to follow clean architecture principles:
- Moved local storage access to Data Layer via AppLocalStorage abstraction
- Applied use case and repository structure for showing onboarding

UI Enhancements:
- Removed hardcoded colors and used colors from AppTheme
- Replaced all hardcoded TextStyles with theme-based styles
- Replaced custom buttons with standard Elevated/OutlinedButtons using themed styles
- Removed static SizedBoxes in buttons, used Expanded for flexibility

 Layout Fixes:
- Made layout fully responsive using Column + Expanded
- Renamed misleading image assets to descriptive names

Splash Screen:
- Removed image-based splash, replaced with native splash using flutter_native_splash

ASHmd97 commented:
@john-safwat 
Thanks John
I applied all your comments.  
Let me know if anything else needs changes.

ASHmd97 committed:
refactor(dependencies): change configureDependencies to return a Future

ASHmd97 committed:
fix(onboarding): update button widths and padding for responsive design

loayelhamaky reviewed (approved):
good job

ASHmd97 committed:
Merge remote-tracking branch 'origin/development' into feature/splash-&-onboarding

ASHmd97 committed:
fix: correct JSON formatting in app_en.arb
